src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts

```diff
@@ -663,7 +663,7 @@ export class ReadingsGraphComponent implements OnDestroy {
 
   create3DGraph(readings: any, ts: any) {
     readings = orderBy(readings, [reading => reading.key.toLowerCase()], ['asc']);
-    const timestamps = ts.map((t: any) => this.dateFormatter.transform(t, 'HH:mm:ss:SSS'));
+    const timestamps = ts.map((t: any) => this.dateFormatter.transform(t, 'HH:mm:ss.SSS'));
```

Comment:
Missing `YYYY-MM-DD` with time format

Reply:
We should probably not show YMD thing for 3D graph!

Reply:
Agree!